### PR TITLE
feat: add Google Gemini API provider (#2)

### PR DESCRIPTION
**Pull request by fanweixiao**
## Related Docs: 

- https://ai.google.dev/docs/gemini_api_overview
- https://ai.google.dev/docs/function_calling

## How to use this provider

```yaml
bridge:
  ai:
    server:
      addr: localhost:8000
      provider: gemini

    providers:
      gemini:
        api_key: <your-api-key>
```

## Be careful that

1. The data format describes in api doc is different from actual api. These can be found from the unit tests.
2. You can not set your function with `-` as this will break your api response. Google will drop characters of the function name in its response. So, I have to check this `-`, and replace to `_` if it presence.

### CODE REVIEW

File: cli/serve.go [hunk 1]
```diff
@@ -112,26 +113,37 @@ var serveCmd = &cobra.Command{
...
+            }
+            log.InfoStatusEvent(os.Stdout, "registered [%s] AI provider", name)
+        case "openai":
+            err := ai.RegisterProvider(openai.NewProvider(provider["api_key"], provider["model"]))
+            if err != nil {
+                return fmt.Errorf("failed to register openai provider: %w", err)
+            }
+            log.InfoStatusEvent(os.Stdout, "registered [%s] AI provider", name)
+        default:
+            log.Warnf("unknown provider: %s", name)
```
Comment (venjiang) [inline cli/serve.go:143]: should be `log.WarningStatusEvent(os.Stdout, "unknown provider: %s", name)`

File: cli/serve.go [hunk 2]
```diff
@@ -112,26 +113,37 @@ var serveCmd = &cobra.Command{
...
+                provider["api_endpoint"],
+                provider["deployment_id"],
+                provider["api_version"],
+            ))
+            if err != nil {
+                return fmt.Errorf("failed to register azopenai provider: %w", err)
+            }
+            log.InfoStatusEvent(os.Stdout, "registered [%s] AI provider", name)
+        case "gemini":
+            err := ai.RegisterProvider(gemini.NewProvider(provider["api_key"]))
```
Comment (venjiang) [inline cli/serve.go:131]: `ai.RegisterProvider` has no return value

**Review by venjiang — CHANGES_REQUESTED** [review]

File: cli/serve.go (hunk 2 above)
Comment (fanweixiao) [reply cli/serve.go]: @venjiang fixed

File: cli/serve.go (hunk 1 above)
Comment (fanweixiao) [reply cli/serve.go]: @venjiang fixed

**Review by venjiang — APPROVED** [review]

**Pull request merged**